v4l2rtspserver-master/src/ImpEncoder.cpp

```diff
@@ -390,10 +395,11 @@ static void *update_thread(void *p) {
             IMP_OSD_UpdateRgnAttrData(prHander[OSD_MOTION], &rAttrDataDetection);
         }
 
-
-        sleep(1);
+        // The timestamp can skip from :00 to :02 with no intermediate :01
+        sleep(0.5);
```

Comment:
0.5 consumes too much CPU, on my Dafang I got from 60% to 80% I am not sure it worth it ....

Reply:
My mistake. 0.5 is rounded down to 0, so the `sleep` call is essentially ignored. I've updated the code to sleep until the next second instead of sleeping for one second, which should solve the issue and remove the negative side effects of the OSD updating at ~1.1s intervals.